mat64/gsvd_example_test.go

```diff
@@ -0,0 +1,89 @@
+// Copyright ©2017 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package mat64_test
+
+import (
+	"fmt"
+	"log"
+	"math"
+
+	"github.com/gonum/matrix"
+	"github.com/gonum/matrix/mat64"
+)
+
+func ExampleGSVD() {
+	// Perform a GSVD factorization on food production/consumption data for the
+	// three years 1990, 2000 and 2014, for Africa and Latin America/Caribbean.
+	//
+	// See Lee et al. doi:10.1371/journal.pone.0030098 and
+	// Alter at al. doi:10.1073/pnas.0530258100 for more details.
+	var gsvd mat64.GSVD
+	ok := gsvd.Factorize(FAO.Africa, FAO.LatinAmericaCaribbean, matrix.GSVDU|matrix.GSVDV|matrix.GSVDQ)
+	if !ok {
+		log.Fatal("GSVD factorization failed")
+	}
+
+	var u, v mat64.Dense
+	u.UFromGSVD(&gsvd)
+	v.VFromGSVD(&gsvd)
+
+	s1 := gsvd.ValuesA(nil)
+	s2 := gsvd.ValuesB(nil)
+
+	fmt.Printf("Africa\n\ts1 = %.4f\n\n\tU = %.4f\n\n",
+		s1, mat64.Formatted(&u, mat64.Prefix("\t    "), mat64.Excerpt(2)))
+	fmt.Printf("Latin America/Caribbean\n\ts2 = %.4f\n\n\tV = %.4f\n",
+		s2, mat64.Formatted(&v, mat64.Prefix("\t    "), mat64.Excerpt(2)))
+
+	var zeroR, q mat64.Dense
+	zeroR.ZeroRFromGSVD(&gsvd)
+	q.QFromGSVD(&gsvd)
+	q.Mul(&zeroR, &q)
+	fmt.Printf("\nCommon basis vectors\n\n\tQ^T = %.4f\n",
+		mat64.Formatted(q.T(), mat64.Prefix("\t      ")))
+
+	// Calculate the antisymmetric angular distances for each eigenvar.
```

Comment:
It made more sense to put it on the types, than on the `Factorize` method, but done.